llvm/lib/Transforms/Utils/LoopUnrollRuntime.cpp

```diff
@@ -940,7 +940,7 @@ bool llvm::UnrollRuntimeLoopRemainder(
     B2.SetInsertPoint(LatchBR);
     auto *Zero = ConstantInt::get(NewIdx->getType(), 0);
     auto *One = ConstantInt::get(NewIdx->getType(), 1);
-    Value *IdxNext = B2.CreateAdd(NewIdx, One, NewIdx->getName() + ".next");
+    Value *IdxNext = B2.CreateNUWAdd(NewIdx, One, NewIdx->getName() + ".next");
```

Comment:
This contradicts the comment just above: 
// Subtle: TestVal can be 0 if we wrapped when computing the trip count,
// thus we must compare the post-increment (wrapping) value."

So I don't believe it holds?

It should hold if we can prove TestVal != 0, but we should already get that in a bunch of places, so unless there's a pass ordering issue, adding it here might not be worthwhile.